site-manage/site-manage-tool/tool/src/java/org/sakaiproject/site/tool/SiteAction.java

```diff
@@ -9749,215 +9762,262 @@ private void actionForTemplate(String direction, int index,
 					    addAlert(state, rb.getString("sitdup.idused") + " ");
 					}
 
-					// duplicated site title is editable; cannot but null/empty after HTML stripping, and cannot exceed max length
-					String titleOrig = params.getString("title");
-					String titleStripped = FormattedText.stripHtmlFromText(titleOrig, true, true);
-					if (isSiteTitleValid(titleOrig, titleStripped, state)) {
-						state.setAttribute(SITE_DUPLICATED_NAME, titleStripped);
+					if (state.getAttribute(STATE_MESSAGE) == null) {
+						// duplicated site title is editable; cannot but null/empty after HTML stripping, and cannot exceed max length
+						String titleOrig = params.getString("title");
+						String titleStripped = FormattedText.stripHtmlFromText(titleOrig, true, true);
+						if (isSiteTitleValid(titleOrig, titleStripped, state)) {
+							state.setAttribute(SITE_DUPLICATED_NAME, titleStripped);
 
-						String newSiteId = null;
-						if (StringUtils.trimToNull(params.getString("newSiteId")) == null) {
-						    newSiteId = IdManager.createUuid();
-						} else{
-						    newSiteId = params.getString("newSiteId");
-						}
-
-						try {
-							String oldSiteId = (String) state
-									.getAttribute(STATE_SITE_INSTANCE_ID);
-							
-							// Retrieve the source site reference to be used in the EventTrackingService
-							// notification of the start/end of a site duplication.
-							String sourceSiteRef = null;
-							try {
-								Site sourceSite = SiteService.getSite(oldSiteId);
-								sourceSiteRef = sourceSite.getReference();
-								
-							} catch (IdUnusedException e) {
-								log.warn(this + ".actionForTemplate; case29: invalid source siteId: "+oldSiteId);
-								return;
+							String newSiteId = null;
```

Comment:
easy opportunity here to get of the else part

Reply:
This code wasn't written as part of this patch, just moved from one location to another. If need be, I can refactor the existing code...